packages/router/src/grpc/view-protocol-server/app-parameters.ts

```diff
@@ -1,10 +1,16 @@
+import { AppParameters } from '@buf/penumbra-zone_penumbra.bufbuild_es/penumbra/core/app/v1/app_pb';
 import type { Impl } from '.';
 import { servicesCtx } from '../../ctx';
 
 export const appParameters: Impl['appParameters'] = async (_, ctx) => {
   const services = ctx.values.get(servicesCtx);
   const { indexedDb } = await services.getWalletServices();
+
+  const subscription = indexedDb.subscribe('APP_PARAMETERS');
   const parameters = await indexedDb.getAppParams();
-  if (!parameters) throw new Error('App parameters not available');
-  return { parameters };
+  if (parameters) return { parameters };
+  for await (const update of subscription)
+    return { parameters: AppParameters.fromJson(update.value) };
```

Comment:
Ah, nice — does this prevent the error that often shows up after clearing the cache that says something about not being able to find AppParameters? If so, this is a great solution!